pyro/infer/mcmc/logger.py

```diff
@@ -38,22 +39,72 @@
 # OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
 
-# TODO: Remove once https://github.com/tqdm/tqdm/issues/650 is resolved.
-class ProgressBar(tqdm):
-    def __init__(self, *args, **kwargs):
-        super(ProgressBar, self).__init__(*args, **kwargs)
+class ProgressBar(object):
+    """
+    Initialize progress bars using :class:`~tqdm.tqdm`.
+
+    :param int warmup_steps: Number of warmup steps.
+    :param int num_samples: Number of MCMC samples.
+    :param int min_width: Minimum column width of the bar.
+    :param int max_width: Maximum column width of the bar.
+    :param int pos: Position of the bar (e.g. in the case of
+        multiple parallel samplers).
+    :param bool disable: Disable progress bar.
+    :param int num_bars: Number of progress bars to initialize.
+        If multiple bars are initialized, they need to be separately
+        updated via the ``pos`` kwarg.
+    """
+    def __init__(self, warmup_steps, num_samples, min_width=80, max_width=120,
+                 disable=False, num_bars=1):
+        total_steps = warmup_steps + num_samples
+        # Disable progress bar in "CI"
+        # (see https://github.com/travis-ci/travis-ci/issues/1337).
+        disable = disable or "CI" in os.environ or "PYTEST_XDIST_WORKER" in os.environ
+        bar_format = None
+        if not ipython_env:
+            bar_format = "{l_bar}{bar}| {n_fmt}/{total_fmt} [{elapsed}, {rate_fmt}{postfix}]"
+        pbar_cls = tqdm_nb if num_bars > 1 and ipython_env else tqdm
```

Comment:
@fehiepsi - This is the line that implements the behavior we discussed in #1728.

Reply:
Got it. Thanks @neerajprad !